### PR TITLE
Proper mechanism for CLI updates added

### LINKED ISSUE #423: CLI is not on path
I installed the desktop but the despite what https://zeit.co/blog/automatic-cli-installation says the cli is never available. I even rebooted. 

When I installed it I was already logged in via a local project install. Is this an edge case? How do I get the CLI to be put on my path? 

For now I'll go back to adding to my project dependencies and launching it with npx

I'm on WIndows 10

### LINKED ISSUE #425: Reverse option order on Windows
This doesn't look good, so let's put "OK" up and "Please, no!" down:

screen shot 2017-12-18 at 03 00 25

Let's also change the text for "Please, no!" 😊

### PR DESCRIPTION
**Pull request by leo**
As of today, we're correctly handling all the cases where the file or directory permissions within the user's environment aren't as we expect them to be.

No more dialogs popping up multiple times! 🌷 

This closes #425 and closes #423 in addition to what's described above.

### CODE REVIEW

File: main/dialogs.js [hunk 1]
```diff
@@ -7,14 +7,22 @@ const { resolve: resolvePath } = require('app-root-path')
 const deploy = require('./utils/deploy')
 
 exports.runAsRoot = (command, why) => {
+  const isWin = process.platform === 'win32'
+  let buttons = ['OK', 'Cancel']
+
+  if (isWin) {
+    buttons = buttons.reverse()
```
Comment (rauchg) [inline main/dialogs.js]: `reverse` mutates the original 😰, so the re-assignment is not needed:

```
> a = [1,2]
[ 1, 2 ]
> a.reverse()
[ 2, 1 ]
> a
[ 2, 1 ]
```

File: main/utils/binary.js [hunk 2]
```diff
@@ -91,13 +91,33 @@ const ensurePath = async () => {
...
+    return
+  }
+
+  try {
+    await fs.chmod(nowPath, mode)
+  } catch (err) {
+    const command = `chmod ${mode} ${nowPath}`
+    const why = 'It needs to make Now CLI executable.'
+
+    // Then move the new binary into position
```
Comment (TooTallNate) [inline main/utils/binary.js]: I think this comment is now incorrect

File: main/utils/binary.js [hunk 3]
```diff
@@ -91,13 +91,33 @@ const ensurePath = async () => {
...
+  // multiple dialog prompts.
+  if (current === mode) {
+    return
+  }
+
+  try {
+    await fs.chmod(nowPath, mode)
+  } catch (err) {
+    const command = `chmod ${mode} ${nowPath}`
+    const why = 'It needs to make Now CLI executable.'
```
Comment (TooTallNate) [inline main/utils/binary.js]: This reads kind of strange. What is "It"? Consider perhaps:

> Now CLI needs permission to be executable

Comment (leo) [reply main/utils/binary.js]: @TooTallNate There's a heading on top + logo next to it both indicating that already. That's why it's not in the description.

File: main/utils/binary.js (hunk 2 above)
Comment (leo) [reply main/utils/binary.js]: Removed!

File: main/dialogs.js (hunk 1 above)
Comment (leo) [reply main/dialogs.js]: Fixed! Thanks 🌊

File: main/utils/binary.js (hunk 3 above)
Comment (rauchg) [reply main/utils/binary.js]: I personally prefer re-iteration for something sensitive like this

File: main/utils/binary.js [hunk 4]
```diff
@@ -208,34 +227,34 @@ exports.getFile = () => {
 
 exports.handleExisting = async next => {
   const destFile = exports.getFile()
+  const parent = path.dirname(destFile)
+  const isWin = process.platform === 'win32'
+
+  const copyPrefix = isWin ? 'copy /b/v/y' : 'cp -p'
+  const copyCommand = `${copyPrefix} ${next} ${destFile}`
+
+  const why = 'It needs to move the downloaded CLI into its place.'
```
Comment (TooTallNate) [inline main/utils/binary.js]: I'd also re-word this one to not use `It` and `its`.

> To move the downloaded CLI into place.

Or maybe

> To place the downloaded CLI in the correct location.

File: main/utils/binary.js [hunk 5]
```diff
@@ -361,11 +380,11 @@ exports.installBundleTemp = async () => {
         try {
           await exports.handleExisting(tempLocation.path)
         } catch (err) {
+          console.error(err)
```
Comment (TooTallNate) [inline main/utils/binary.js]: Are these leftover or intentional?

Comment (leo) [reply main/utils/binary.js]: Yep 👍

File: main/utils/binary.js (hunk 4 above)
Comment (leo) [reply main/utils/binary.js]: 👍 done

File: main/utils/binary.js (hunk 3 above)
Comment (leo) [reply main/utils/binary.js]: @TooTallNate suggested a better text below 🚀

Comment (TooTallNate) [reply main/utils/binary.js]: That was for a different instance though. I still think this line needs updating.

Comment (leo) [reply main/utils/binary.js]: @TooTallNate Done!

**Pull request merged**